### PR TITLE
fix(rest): fix document upload in SaaS, using multipart requests

### PR DESCRIPTION
**Pull request by johnBgood**
## Description

Uploading documents using the REST Connector and a `multipart/form-data` request is now working correctly in SaaS (it was working in SM).
- The cluster will convert the Documents to a new custom `CloudFunctionFilePart` type. This is required so that all the information (filename, content type) is serialized and sent to the CloudFunction.
- Previously only the Document content was sent, leading to the request payload not being identified as a file.

## Checklist

- [x] PR has a **milestone** or the `no milestone` label.

### CODE REVIEW

File: connectors/http/http-base/src/main/java/io/camunda/connector/http/base/cloudfunction/CloudFunctionFilePart.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *
+ * @param name The name of the field in the multipart form corresponding to this part.
+ * @param submittedFileName If this part represents an uploaded file, gets the file name submitted
+ *     in the upload. Returns {@code null} if no file name is available or if this part is not a
+ *     file upload.
+ * @param content The content of this part.
+ * @param contentType The content type of this part.
+ */
+public record CloudFunctionFilePart(
+    String name, String submittedFileName, byte[] content, String contentType) {}
```
Comment (sbuettner) [inline connectors/http/http-base/src/main/java/io/camunda/connector/http/base/cloudfunction/CloudFunctionFilePart.java:32]:
```suggestion
    String name, String fileName, byte[] content, String contentType) {}
```

**Review by chillleader — DISMISSED** [review]
Perfect 🚀

**Comment by johnBgood** [comment]
# QA information
## Test Environment
- Test required in (you can check both):
  - [ ] SaaS
  - [ ] SM
- [ ] Create a new test for this feature in our [Pre Release Tests](https://modeler.ultrawombat.com/projects/017779a8-9459-4feb-8b47-eeef8b2717ef--connectors-pre-release-test) for @Szik to be able to test

## Test Scope
*Please describe the test scope, happy path, edge cases that come to your mind, and whatever you think might relevant to test*

## Test Data
*Please provide the test data, if needed (files, URLs, code snippets, FEEL expressions)*
